File: BackEnd - ControleDeVacinacaoBovina/Models/RegistroVacinacao.cs
using System;

namespace ControleDeVacinacaoBovina.Models
{
    public class RegistroVacinacao
    {
        public int IdRegistroVacinacao { get; set; }
        public int Quantidade { get; set; }
        public DateTime DataDaVacina { get; set; }
        public int IdRebanho { get; set; }
        public int IdVacina { get; set; }
        public Rebanho Rebanho { get; set; }
        public Vacina Vacina { get; set; }
        public DateTime DataDeCadastro { get; set; } = DateTime.Now;
        public bool Ativo { get; private set; } = true;
        public void SetAtivo(bool ativo)
        {
            Ativo = ativo;
        }
    }
}

File: FrontEnd - ControleDeVacinacaoBovina/angular-controle-de-vacinacao-bovina/src/Entities/registrovacinacao.ts
import { Animal } from "./animal";
import { Vacina } from "./vacina";

export interface RegistroVacina{
  IdRegistroVacinacao: number,
  Quantidade: number,
  DataDaVacina: Date,
  IdAnimal: number,
  IdVacina: number,
  Animal: Animal,
  Vacina: Vacina,
  Ativo: boolean
}
